src/Server.zig

```diff
@@ -469,6 +470,101 @@ fn getAstCheckDiagnostics(
     }
 }
 
+/// TODO: Memory
+pub fn getFullCompilationDiagnostics(
+    server: *Server,
+    handle: DocumentStore.Handle,
+    diagnostics: *std.ArrayListUnmanaged(types.Diagnostic),
+) !void {
+    var allocator = server.arena.allocator();
+    const config = server.config;
+
+    const build_file_uri = handle.associated_build_file orelse return;
+    const build_file = server.document_store.build_files.get(build_file_uri).?;
+
+    const tracy_zone = tracy.trace(@src());
+    defer tracy_zone.end();
+
+    var arena = std.heap.ArenaAllocator.init(allocator);
+    defer arena.deinit();
+    const arena_allocator = arena.allocator();
+
+    const build_file_path = try uri_utils.parse(arena_allocator, build_file.uri);
+    const directory_path = try std.fs.path.resolve(arena_allocator, &.{ build_file_path, "../" });
+
+    // TODO extract this option from `BuildAssociatedConfig.BuildOption`
+    const zig_cache_root: []const u8 = try std.fs.path.join(arena_allocator, &.{ directory_path, "zig-cache" });
```

Comment:
it would be great if zig supported outputting errors as json

Reply:
That'll never happen - Andrew hates JSON.

Reply:
There is a proposal for it https://github.com/ziglang/zig/issues/5456

Reply:
Maybe there *is* hope :P (wonder if it'll ever get accepted though)